scripts/js/groups-domains.js

```diff
@@ -496,15 +496,17 @@ function addDomain() {
     return;
   }
 
-  for (var i = 0; i < domains.length; i++) {
-    if (kind === "exact" && wildcardChecked) {
+  // Check if the wildcard checkbox was marked and transform the domains into regex
+  if (kind === "exact" && wildcardChecked) {
+    for (var i = 0; i < domains.length; i++) {
       // Transform domain to wildcard if specified by user
       domains[i] = "(\\.|^)" + domains[i].replaceAll(".", "\\.") + "$";
-      kind = "regex";
 
       // strip leading "*." if specified by user in wildcard mode
       if (domains[i].startsWith("*.")) domains[i] = domains[i].substr(2);
     }
```

Comment:
Looking at the code, I can see this:

If the user enters `*.example.com` the domain will be change into a wildcard (line 503) **before** the `*.` is removed (line 506), resulting in this regex: `(\.|^)*\.example.com`. 
Is this intended?

The question is: `*.example.com` should be changed into what regex?

- `(\.|^)example.com$` (every subdomains and also the main domain);
- `\.example.com$` (only subdomains, but not the main domain);
- `(\.|^)*\.example.com$` (same as above, but more complex to parse);

Reply:
> Is this intended?

I don't think so. I think the order here is wrong. The removal of `*.` was meant as a help because users entered domains like this and checked the wildcard box because they think they had to enter `*.` for it to become a wildcard.

> The question is: *.example.com should be changed into what regex?


I think 

> (\.|^)example\.com$ (every subdomains and also the main domain);

Reply:
Then I will change the order and remove the `*.` text before turning it into a regex.

Reply:
In v5, it was explicitly the latter: all subdomains but **not** the domain itself. This is rather uncritical and non-breaking here as the checkbox is just a assistance tool for those inexperienced with regex. To me, `*.abc.com` does **clearly** not include `abc.com` itself, however, this is just me (and Pi-hole v5). You are perfectly right to ask "what would the user *expect* to happen". I don't know TBH. But @yubiuser may be right (include also the domain itself).

Reply:
We can discuss this a little more before we decide. I don't want to change the previous behavior unless we have a good reason.

Reply:
I just tested this on v5 and v6. Entering `test.com` produced `(\.|^)test\.com$` in **both** version. So nothing would change.

Reply:
This is the relevant code section
https://github.com/pi-hole/web/blob/be05b0f61d3fcf796dae4cc3f89f8540b2359325/scripts/pi-hole/php/groups.php#L618-L620

Reply:
If we want to make it super clear to users we could change

https://github.com/pi-hole/web/blob/42e7279ad5c9495058430be2cbdb2da010f43cef/groups-domains.lp#L63

'....to involve all subdomains' >> 'to involve the domain and all subdomains.'

Reply:
> Entering `test.com` produced `(\.|^)test\.com$` in both version. 

My doubt was only about the "old" format, where users insert domains with an wildcard, like `*.test.com`. 

The original code (before this PR) wasn't removing the leading `*.`. 
Now it is working as expected even if an user uses this format.

I think this is ready to review.